x-pack/platform/plugins/shared/lens/public/editor_frame_service/editor_frame/frame_layout.tsx

```diff
@@ -40,21 +49,58 @@ export function FrameLayout(props: FrameLayoutProps) {
           </aside>
         </EuiFlexItem>
       ) : null}
-      <EuiFlexItem grow={true} className="lnsFrameLayout__wrapper">
+      <EuiFlexItem
+        grow={true}
+        css={css`
+          position: relative;
+        `}
+      >
         <EuiPage
           paddingSize="none"
-          className={classNames('lnsFrameLayout', {
-            'lnsFrameLayout-isFullscreen': isFullscreen,
-          })}
+          css={css`
+            padding: 0;
+            position: absolute;
+            top: 0;
+            left: 0;
+            right: 0;
+            bottom: 0;
+            overflow: hidden;
+            flex-direction: column;
+            ${euiBreakpoint(euiThemeContext, ['xs', 's', 'm'])} {
+              position: static;
+            }
+          `}
         >
           <EuiPageBody
             restrictWidth={false}
             className="lnsFrameLayout__pageContent"
             aria-labelledby="lns_ChartTitle"
+            css={css`
+              overflow: hidden;
+              flex-grow: 1;
+              flex-direction: row;
+              ${euiBreakpoint(euiThemeContext, ['xs', 's', 'm'])} {
+                flex-wrap: wrap;
+                overflow: auto;
+                > * {
+                  flex-basis: 100%;
+                }
+              }
+            `}
           >
             <section
-              className={'lnsFrameLayout__sidebar lnsFrameLayout__sidebar--left hide-for-sharing'}
+              className={'hide-for-sharing'}
               aria-labelledby="dataPanelId"
+              css={[
+                sidebarStyles(euiThemeContext),
+                css`
+                  ${isFullscreen &&
+                  `
+                // Hide the datapanel in fullscreen mode. Using display: none does trigger
+                // a rerender when the container becomes visible again, maybe pushing offscreen is better
+                display: none;`}
+                `,
+              ]}
```

Comment:
suggestion: you can avoid using an interpolation here, I think it looks clearer but feel free to leave it as it is :)

```suggestion
              css={[
                sidebarStyles(euiThemeContext),
                isFullscreen &&
                  css`
                    display: none;
                  `,
              ]}
```

Reply:
dfeefcdb5eed554955b78664d978cc8d1bb3663d